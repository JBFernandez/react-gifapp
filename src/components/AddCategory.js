import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState("");

    const handleInputChange = (e) => {        
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('handleSubmit', inputValue);
        //we make the callback to setCategories
        //we could also send the categories as a prop

        if (inputValue.trim().length > 2) {
            setCategories( cats => [inputValue, ...cats]);
            setInputValue("");
        }
    }

  return (
        <form onSubmit={ handleSubmit }> {/*form would be the grouping element, we dont need the fragment */}
            <p> { inputValue } </p>
            <input
                type="text"
                value={inputValue}
                onChange={ handleInputChange } //e is the event
            /> {/* input */}
        </form>
  )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
