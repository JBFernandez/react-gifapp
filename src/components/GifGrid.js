import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import GifGridItem from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {
  
  const { data: images, loading } = useFetchGifs( category );
  console.log(loading);

//with async, getGifs returns a promise
  

  return (
    <>
    <h3 className='animate__animated animate__fadeIn'> { category } </h3>
    {loading && <p className='animate__animated animate__flash'>Loading</p>}
    
    <div className='card-grid'>     
          {
            images.map( (img) => {
             return <GifGridItem
                      key={img.id}
                      { ...img } 
                    />             
            })
          }        
    </div>
    </>
  )
}

GifGrid.propTypes = {}

export default GifGrid