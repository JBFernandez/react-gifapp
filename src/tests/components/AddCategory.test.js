import {shallow} from 'enzyme';
import { AddCategory } from '../../components/AddCategory';
import React from 'react';
import '@testing-library/jest-dom';


describe('Tests in the <AddCategory /> component', () => { 

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={setCategories}/> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={setCategories}/> );
    } )
    
    test('should display correctly', () => { 

        expect(wrapper).toMatchSnapshot();
     })

     test('should change the text box', () => { 

        const input = wrapper.find('input');
        const value = 'Hello World'

        input.simulate('change', { target: {
            value: value
        } 
    });

        expect(wrapper.find('p').text().trim()).toBe(value);

      })

      test('should not post information on submit', () => { 
          wrapper.find('form').simulate('submit', { preventDefault: () => {} })

          expect( setCategories ).not.toHaveBeenCalled();
       })

       test('should call setCategories and clear the text box', () => {
           const value = 'AOT'; 
           wrapper.find('input').simulate('change', { target: {
               value: value
           } })
           expect( wrapper.find('p').text().trim() ).toBe( value );

           wrapper.find('form').simulate('submit', { preventDefault: () => {}, })
        

           expect( setCategories ).toHaveBeenCalled();
           expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
           expect( wrapper.find('input').prop('value') ).toBe('');

           

        })//test clear box
    
 })