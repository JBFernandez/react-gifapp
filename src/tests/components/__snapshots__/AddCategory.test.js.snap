// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in the <AddCategory /> component should display correctly 1`] = `
<form
  onSubmit={[Function]}
>
   
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
   
</form>
`;
